ui/src/message_popup/author/edit_question_page.jsx

```diff
@@ -11,15 +11,96 @@ export default React.createClass({
   displayName: 'EditQuestionPage',
 
   propTypes: {
-    questionId: React.PropTypes.string.isRequired
+    questionId: React.PropTypes.string.isRequired,
+    loaded: React.PropTypes.bool,
+    allQuestions: React.PropTypes.object
+  },
+
+  getDefaultProps(){
```

Comment:
This is a good idea based on our conversation yesterday, but I think for this scenario we'll be better served by mocking the calls to the server.  As an example, in test we could let this component make calls through the Api module and use [sinon](http://sinonjs.org/docs/) to spy on that.  Or we can allow passing in an instance of `api` as a prop, and then pass in one that has all spies in test.

This seems like a better approach to me for this particular component, because the essence of what it is doing is making those calls to the API, and so if we have a test that passes in props then we don't test any of that code.

So I'd suggest removing these props, and then if this seems in good shape we can also merge this and add in additional tests in a separate pass (since this feature isn't in production use yet).